src/v/raft/state_machine_manager.cc

```diff
@@ -433,10 +443,13 @@ ss::future<> state_machine_manager::background_apply_fiber(
         try {
             model::record_batch_reader reader = co_await _raft->make_reader(
               config);
-            co_await std::move(reader).consume(
+            auto last_applied_before = entry->stm->last_applied_offset();
+            auto last_applied_after = co_await std::move(reader).consume(
               batch_applicator(background_ctx, {entry}, _as, _log),
               model::no_timeout);
-
+            if (last_applied_before >= last_applied_after) {
```

Comment:
can it only happen when nothing has been applied and `last_applied_after` is `model::offset{}`?

Reply:
you are right, i need to change that, great find